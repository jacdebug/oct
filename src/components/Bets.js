import React from 'react';
import styled from 'styled-components';
import Box from './Box';

const Bets = ({ bets }) =>
  <Box>
    {bets.map((bet, index) =>
      <BetsWrapper key={index}>
        <div>
          {bet.name}
        </div>
        <div className="odds">
          {bet.bestOddsDecimal}
        </div>
      </BetsWrapper>
    )}
  </Box>;

export default Bets;

const BetsWrapper = styled.div`
  flex-grow: 1;
  border: 10px solid ${props => props.theme.tcolor1};
  border-width: 10px 0 10px 10px;
  text-align: center;

  div {
    margin: 5px;
  }

  .odds {
    color: ${props => props.theme.tcolor3};
  }

  &:last-child {
    border-width: 10px;
  }
`;
