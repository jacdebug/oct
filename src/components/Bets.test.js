import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Bets from './Bets';

const testData = [
  {
    name: 'Barcelona',
    genericName: 'HOME',
    bestOddsDecimal: 1.05,
  },
  {
    name: 'Empate',
    genericName: 'DRAW',
    bestOddsDecimal: 15,
  },
  {
    name: 'Granada',
    genericName: 'AWAY',
    bestOddsDecimal: 51,
  },
];

it('renders without crashing', () => {
  shallow(<Bets bets={testData} />);
});

it('renders a snapshot', () => {
  const tree = shallow(<Bets bets={testData} />);
  expect(toJson(tree)).toMatchSnapshot();
});

it('renders list of bets', () => {
  const wrapper = shallow(<Bets bets={testData} />);
  expect(wrapper.find('.odds')).toHaveLength(3);
});
