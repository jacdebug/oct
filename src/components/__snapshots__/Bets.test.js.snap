// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<styled.div>
  <styled.div>
    <div>
      Barcelona
    </div>
    <div
      className="odds"
    >
      1.05
    </div>
  </styled.div>
  <styled.div>
    <div>
      Empate
    </div>
    <div
      className="odds"
    >
      15
    </div>
  </styled.div>
  <styled.div>
    <div>
      Granada
    </div>
    <div
      className="odds"
    >
      51
    </div>
  </styled.div>
</styled.div>
`;
